src/openrct2/ride/Vehicle.cpp

```diff
@@ -9312,94 +9309,81 @@ int32_t Vehicle::UpdateTrackMotionMiniGolf(int32_t* outStation)
         vehicle = GET_VEHICLE(nextVehicleIndex);
     }
 
-    regs.eax /= regs.ebx;
-    regs.ecx = (regs.eax * 21) >> 9;
-    regs.eax = velocity >> 12;
-    regs.ecx -= regs.eax;
-    regs.ebx = velocity;
-    regs.edx = velocity >> 8;
-    regs.edx *= regs.edx;
-    if (regs.ebx < 0)
+    int32_t newAcceleration = ((sumAcceleration / numVehicles) * 21) >> 9;
+    newAcceleration -= velocity >> 12;
```

Comment:
This does not appear to be correct. In the old situation, newacceleration was taken from (velocity >> 12) and you swapped that around here.

Reply:
newAcceleration is regs.ecx, and I removed regs.eax

It does seem confusing. Would you like me to have an auxiliary variable for regs.eax?

Reply:
I thought it looked fine. Isn't it also the same as the non golf version of this function